core/src/main/java/org/apache/struts2/dispatcher/multipart/AbstractMultiPartRequest.java

```diff
@@ -54,6 +54,12 @@ public abstract class AbstractMultiPartRequest implements MultiPartRequest {
     protected long maxSize;
     protected boolean maxSizeProvided;
 
+    /**
+     * Specifies the maximum number of files in one request.
+     */
+    protected long maxFiles;
+    protected boolean maxFilesProvided;
```

Comment:
nitpick: can't we use Long instead of long here? and a null value will mean that it's not provided

Reply:
I have used the same approach as with `maxSize`, but maybe you are right to use `null` as a _not set_ indicator

Reply:
Done